packages/components/src/components/SignIn.vue

```diff
@@ -42,21 +120,108 @@
 
 <script>
 import AppMapLogo from '@/assets/appmap-full-logo.svg';
-import shieldIcon from '@/assets/shield-icon.svg';
-import VExternalLinkIcon from '@/assets/external-link.svg';
+import ShieldIcon from '@/assets/shield-icon.svg';
+import GitHubLogo from '@/assets/github-logo.svg';
+import GitLabLogo from '@/assets/gitlab-logo.svg';
+import EmailIcon from '@/assets/email.svg';
+import ExternalLinkIcon from '@/assets/external-link.svg';
+
+const GENERIC_ERROR_MSG = 'Something went wrong, please try again later.';
 
 export default {
   name: 'v-sign-in',
+
   components: {
     AppMapLogo,
-    shieldIcon,
-    VExternalLinkIcon,
+    ShieldIcon,
+    GitHubLogo,
+    GitLabLogo,
+    ExternalLinkIcon,
+    EmailIcon,
+  },
+
+  props: {
+    appmapServerUrl: {
+      type: String,
+      default: 'https://getappmap.com',
+    },
+  },
+
+  data() {
+    return {
+      email: '',
+      submitted: false,
+      verificationCode: '',
+      error: '',
+    };
   },
 
   methods: {
+    keyDown(event) {
+      if (event.key === 'Enter') {
+        if (event.srcElement.id === 'email-input') {
+          this.activateWithEmail();
+        } else if (event.srcElement.id === 'verification-code-input') {
+          this.completeActivation();
+        }
+      }
+    },
     signIn() {
       this.$root.$emit('sign-in');
     },
+    reset() {
+      this.email = '';
+      this.error = '';
+      this.verificationCode = '';
+      this.submitted = false;
+    },
+    async activateWithEmail() {
+      const url = new URL('/api/activations', this.appmapServerUrl);
+      this.addParamsToUrl(url, { email: this.email });
+      const response = await fetch(url, { method: 'POST' });
+
+      if (response.status === 201) {
+        this.error = '';
+        this.submitted = true;
+      } else if (response.status < 500 && response.status >= 400) {
+        this.email = '';
+        this.error = this.generateInvalidFieldMsg('email address');
+      } else {
+        this.email = '';
+        this.error = GENERIC_ERROR_MSG;
+      }
+    },
+    async completeActivation() {
+      const url = new URL('/api/activations/verify', this.appmapServerUrl);
+      this.addParamsToUrl(url, {
+        email: this.email,
+        code: this.verificationCode,
+      });
+      const response = await fetch(url, { method: 'POST' });
+
+      if (response.status === 200) {
+        try {
+          const data = await response.json();
+          this.$root.$emit('activate', data.api_key);
+        } catch (error) {
+          this.submitted = false;
+          this.error = GENERIC_ERROR_MSG;
+        }
+      } else {
+        this.verificationCode = '';
+        this.submitted = false;
```

Comment:
If the user inputs the verification code incorrectly then by changing this flag we're forcing them to request another as there's no way to get back to the second screen. Can we present the error on the second screen? It looks like there's already a 'try again' button if they need to go back.

Reply:
Yeah, good point, I made it so that the error is on the second screen and it doesn't go back to the first screen :+1: